llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -910,7 +950,9 @@ bool modifyToS20(InstrNode Start, MachineRegisterInfo &MRI, MachineIRBuilder &B,
   switch (StartNodeMI->getOpcode()) {
   case TargetOpcode::COPY:
   case TargetOpcode::G_LOAD:
-  case TargetOpcode::G_PHI: {
+  case TargetOpcode::G_PHI:
+  case AIE2::G_AIE_ZEXT_EXTRACT_VECTOR_ELT:
+  case AIE2::G_AIE_SEXT_EXTRACT_VECTOR_ELT: {
```

Comment:
We should somehow abstract this to arch specific files. As @andcarminati  mentioned we want all our combiner passes to work generically.

Reply:
The good thing: we already have this in `TII->getGenericExtractVectorEltOpcode`

Reply:
When I attempt to replace AIE2::G_AIE_Z/SEXT_EXTRACT_VECTOR_ELT with TII->getGenericExtractVectorEltOpcode(SignExt), I encounter an issue because the latter returns a dynamic value, which cannot be used directly in a switch case label.

Reply:
I suggest to concentrate those cases in the default case (with ifs), because is is important to have generic code here.